src/fullscreen.js

```diff
@@ -58,6 +63,11 @@ class FullScreen {
             document.body.classList.add('dplayer-web-fullscreen-fix');
 
             this.player.events.trigger('webfullscreen');
+
+            if (this.isFullScreen('browser')) {
+                this.cancel('browser');
```

Comment:
I think **request()** should only do **fullscreen** logically. I recommend to move them to **toggle()**:

```js
     toggle (type = 'browser') {
         if (this.isFullScreen(type)) {
             this.cancel(type);
         }
         else {
             this.request(type);
             // make fullscreen state mutex here.
         }
      }
```